### PR TITLE
Lead School to Lead Partner Part IV

### PR DESCRIPTION
**Pull request by adamjcolvin**
### Context
[Trello](https://trello.com/c/YlTKATXq)

**Related PRs**
https://github.com/DFE-Digital/register-trainee-teachers/pull/4527 https://github.com/DFE-Digital/register-trainee-teachers/pull/4541 https://github.com/DFE-Digital/register-trainee-teachers/pull/4551

This continues the refactoring of lead_school references throughout the app to lead_partner.  

### Changes proposed in this pull request

The main focus of this PR is on removing the `lead_school` column from the schools table. This involved changing the model itself, as well as factories, specs and any remaining code that still has a notion of a school being a lead school. 

### Guidance to review

For now, I've left the `lead_school` column on the trainees table, just so this PR doesn't get too big. But I think a sensible next step would be to remove that column, and the associated `lead_school_trainees` association on the schools table. 

### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

**Review by kennyevil — APPROVED** [review]

File: app/services/trainees/create_from_hesa.rb [hunk 1]
```diff
@@ -138,10 +138,10 @@ def lead_partner_attributes
 
       if hesa_trainee[:ukprn].in?(LEAD_PARTNER_TO_ACCREDITED_PROVIDER_MAPPING.keys)
         attrs.merge!(lead_partner: LeadPartner.find_by(ukprn: hesa_trainee[:ukprn]), lead_partner_not_applicable: false)
-      elsif hesa_trainee[:lead_school_urn].in?(NOT_APPLICABLE_SCHOOL_URNS)
+      elsif hesa_trainee[:lead_partner_urn].in?(NOT_APPLICABLE_SCHOOL_URNS)
```
Comment (tomtrentham) [inline app/services/trainees/create_from_hesa.rb:141]: Just a note about the timing of changes to this HESA process and the HESA testing that's currently in progress. Nothing to action, but wanted us to be aware of it.

File: spec/services/funding/lead_partner_payment_schedules_importer_spec.rb [hunk 2]
```diff
@@ -170,10 +170,10 @@ module Funding
         }
       end
 
-      let(:lead_school_one) { create(:school, :lead, urn: "131238") }
-      let(:lead_school_two) { create(:school, :lead, urn: "135438") }
-      let(:lead_school_three) { create(:school, :lead, urn: "103527") }
-      let(:lead_school_four) { create(:school, :lead, urn: "105491") }
+      let(:lead_school_one) { create(:school, urn: "131238") }
```
Comment (tomtrentham) [inline spec/services/funding/lead_partner_payment_schedules_importer_spec.rb:173]: Should we still be using `lead_school` terminology?

File: app/components/system_admin/schools/details/view.rb [hunk 3]
```diff
@@ -12,7 +12,7 @@ def initialize(school_entity)
 
         def lead_partner?
           if school_entity.is_a?(School)
-            school_entity.lead_school
+            school_entity.lead_partner?
           else
             school_entity.lead_partner
           end
```
Comment (tomtrentham) [inline app/components/system_admin/schools/details/view.rb:18]: Is it just me, or does this whole method look a bit strange now? 🤔 Could we neaten this up at all?

**Review by tomtrentham — COMMENTED** [review]
A couple of small questions inline.

File: app/components/system_admin/schools/details/view.rb (hunk 3 above)
Comment (adamjcolvin) [reply app/components/system_admin/schools/details/view.rb]: Good shout, I added an alias to the school form called lead_partner?, so now the method name is the same in both cases. So now we can get rid of this completely and replace it with `delegate :lead_partner?, to: :school_entity`. See the commit [here](https://github.com/DFE-Digital/register-trainee-teachers/pull/4577/commits/636061bfc23c2cafc858783f60214e0fea3f07f2).

**Review by tomtrentham — APPROVED** [review]

**Pull request merged**